### PR TITLE
Adding a seed file to lzo_decompress_target_seed

### PR DESCRIPTION
**Pull request by viniul**
This commit adds a minimal lzo seed as a seed for the lzo_decompress_target. Still results in a heap-buffer-overflow at the moment.

### CODE REVIEW

**Comment by viniul** [comment]
Adding a seed for lzo_decompress_target. Seed immediately leads to crash on current harness. I suspect something is buggy with the harness itself - I will look into that:
md5sum build/out/lzo/./crash-67c0a79406c899cf7aa995b5fa7e6a7e89140128
539ab8da41409a95e55746bf55d42a9a  build/out/lzo/./crash-67c0a79406c899cf7aa995b5fa7e6a7e89140128

md5sum projects/lzo/lzo_decompress_target_seeds/seed.lzo 
539ab8da41409a95e55746bf55d42a9a  projects/lzo/lzo_decompress_target_seeds/seed.lzo

Seed file is taken from american fuzzy lop repo

**Comment by viniul** [comment]
I think lzo1x_decompress might not be a good target - Taking a look at this manual:
http://manual.freeshell.org/lzo/LZO.FAQ

The crashing seems to be expected sort of:

- lzo1x_decompress
    The 'standard' decompressor. Pretty fast - use this whenever possible.

    This decompressor expects valid compressed data.
    If the compressed data gets corrupted somehow (e.g. transmission
    via an erroneous channel, disk errors, ...) it will probably crash
    your application because absolutely no additional checks are done.

- lzo1x_decompress_safe
    The 'safe' decompressor. Somewhat slower.

    This decompressor will catch all compressed data violations and
    return an error code in this case - it will never crash.

File: projects/lzo/Dockerfile [hunk 1]
```diff
@@ -20,3 +20,4 @@ RUN apt-get update && apt-get install -y make autoconf automake libtool wget
 RUN wget -O lzo.tar.gz \
     http://www.oberhumer.com/opensource/lzo/download/lzo-2.10.tar.gz
 COPY *.c *.options build.sh $SRC/
+COPY lzo_decompress_target_seeds $SRC/lzo_decompress_target_seeds
```
Comment (bshastry) [inline projects/lzo/Dockerfile:23]: Thanks @viniul . I think if it's just one seed, we can skip it. However, if there are more, we can check it into another github repo and fetch them here via something like

```
RUN wget <url>
```

**Comment by bshastry** [comment]
> lzo1x_decompress_safe
The 'safe' decompressor. Somewhat slower.

> This decompressor will catch all compressed data violations and
> return an error code in this case - it will never crash.

Thanks @viniul . I changed `lzo1x_decompress` to `lzo1x_decompress_safe` now but it crashes on an assertion failure (I am not sure if the assertion condition is legit).

Will debug this further.

**Comment by viniul** [comment]
Doing the same at the moment. I think we should remove the assertion  since return value of lzo1x_decompress_safe is the error code. So assertion would trigger each time decompress_safe receives non valid lzo file - which will happen of course, if we are fuzz testing.

File: projects/lzo/Dockerfile (hunk 1 above)
Comment (viniul) [reply projects/lzo/Dockerfile:23]: Do you think it makes sense to build up a small seed corpus of multiple lzo files?

**Comment by bshastry** [comment]
Minor update: After removing the assertion and switching to the safe decompressor API, the fuzzer works. I am currently checking if this seed improves coverage. If it does, I will merge this. Otherwise, let's keep it on hold for now.

**Comment by bshastry** [comment]
Merged manually. Seed slightly improves coverage.

**Pull request merged**